tools/clang/tools/hlsl2spirv/hlsl2spirv.cpp

```diff
@@ -0,0 +1,216 @@
+  IFT(pLibrary->CreateIncludeHandler(&pIncludeHandler));
+  IFT(dllSupport.CreateInstance(CLSID_DxcCompiler, &pCompiler));
+  IFT(pCompiler->Compile(pSource, inputFileName, entry, profile, flags.data(),
+                         flags.size(), nullptr, 0, pIncludeHandler, &pResult));
+  // Get compilation results.
+  IFT(pResult->GetStatus(&resultStatus));
+
+  // Get diagnostics string and print warnings and errors to stderr.
+  IFT(pResult->GetErrorBuffer(&pErrorBuffer));
+  const std::string diagnostics((char *)pErrorBuffer->GetBufferPointer(),
+                                pErrorBuffer->GetBufferSize());
+  fprintf(stderr, "%s\n", diagnostics.c_str());
+
+  if (SUCCEEDED(resultStatus)) {
+    IFT(pResult->GetResult(&binaryBlobPtr));
+  } else {
+    throw("Compilation of the input HLSL via dxc failed.");
+  }
+}
+
+int __cdecl wmain(int argc, const wchar_t **argv_) {
+  const char *pStage = "Operation";
+  try {
+    pStage = "Argument processing";
+
+    // Parse command line options.
+    const OptTable *optionTable = getHlslOptTable();
+    MainArgs argStrings(argc, argv_);
+    DxcOpts dxcOpts;
+    DxcDllSupport dxcSupport;
+    dxc::EnsureEnabled(dxcSupport);
+
+    // TODO: We currently do not have an options table for hlsl2spirv.
+    // Therefore we have used the DxcOpts class to handle command line
+    // arguments.
+    // Ideally this must be fixed by:
+    // 1- hlsl2spiv having its own Hlsl2SpirvOpts
+    // 2- Supporting only the options we actually need (and erroring out for any
+    // other option).
+
+    // Read options and check errors.
+    {
+      std::string errorString;
+      llvm::raw_string_ostream errorStream(errorString);
+      int optResult =
+          ReadDxcOpts(optionTable, DxcFlags, argStrings, dxcOpts, errorStream);
+      errorStream.flush();
+      if (errorString.size()) {
+        fprintf(stderr, "hlsl2spirv failed : %s", errorString.data());
+      }
+      if (optResult != 0) {
+        return optResult;
+      }
+    }
+
+    // Handle help request, which overrides any other processing.
+    if (dxcOpts.ShowHelp) {
+      std::string helpString;
+      llvm::raw_string_ostream helpStream(helpString);
+      optionTable->PrintHelp(helpStream, "hlsl2spirv.exe",
+                             "HLSL to SPIR-V Compiler");
+      helpStream.flush();
+      WriteUtf8ToConsoleSizeT(helpString.data(), helpString.size());
+      return 0;
+    }
+
+    // Apply defaults.
+    if (dxcOpts.EntryPoint.empty() && !dxcOpts.RecompileFromBinary) {
+      dxcOpts.EntryPoint = "main";
+    }
+
+    Hlsl2SpirvContext context(dxcSupport, dxcOpts);
+    pStage = "Compilation";
+    context.Compile();
+    context.ReportResults();
+  } catch (const ::hlsl::Exception &hlslException) {
+    try {
+      const char *msg = hlslException.what();
+      // printBuffer is safe to treat as UTF-8 because we use ASCII errors only.
+      Unicode::acp_char printBuffer[128];
```

Comment:
128 seems low.   Do we really have to use a fixed size buffer?
Might as well make it 8K or something.